kong/client.go

```diff
@@ -164,6 +167,29 @@ func NewClient(baseURL *string, client *http.Client) (*Client, error) {
 	return kong, nil
 }
 
+// SetWorkspace sets the Kong Enteprise workspace in the client.
+// Calling this function with an empty string resets the workspace to default workspace.
+func (c *Client) SetWorkspace(workspace string) {
+	c.workspaceLock.Lock()
+	defer c.workspaceLock.Unlock()
+	c.workspace = workspace
+}
+
+//Workspace return the workspace
+func (c *Client) Workspace() string {
+	c.workspaceLock.RLock()
+	defer c.workspaceLock.RUnlock()
+	return c.workspace
+}
+
+// baseURL build the base URL from the rootURL and the workspace
+func (c *Client) baseURL(workspace string) string {
```

Comment:
```suggestion
func (c *Client) workspacedBaseURL(workspace string) string {
```
Makes it a little more descriptive and self-documenting.
@mflendrich Thoughts?

Reply:
Done